lib/rawlink_stubs.c

```diff
@@ -271,40 +264,58 @@ af_packet_setfilter(int fd, value vfilter)
 	return (r);
 }
 
+int af_packet_set_promisc(int fd, int promisc, char *ifname)
+{
+	struct ifreq ifopts;
+	int r;
+	if (!promisc)
+		return (1);
+
+	strncpy(ifopts.ifr_name, ifname, IFNAMSIZ - 1);
+	caml_enter_blocking_section();
+	r = ioctl(fd, SIOCGIFFLAGS, &ifopts);
```

Comment:
You have to test the error of each syscall and abort, also ifopts should be bzroed before.